### ISSUE TITLE
Error attempting to create app package using Windows Application Package project type (desktop bridge)

### ISSUE THREAD

**Issue by nathan-alden-hp** [dotnet/core#3137]
# Issue Title

A build error is returned attempting to build a Windows Application Package project.

# Tool Versions

Visual Studio Enterprise 2019 Preview 16.3.0 Preview 1.0
.NET Core 3.0.100-preview7-012821

# General

I'm not sure if this issue is with Visual Studio, MSBuild, or .NET Core, so I'm posting here.

I've recently been testing out creating a so-called "desktop bridge" project for a WPF app I'm writing. Everything seems to work except that I get a build error when I try and create an app package. The entire solution compiles just fine; it's only the app package creation that's failing:
```
1>------ Publish started: Project: WindowsApplicationPackage, Configuration: Debug x64 ------
1>Restore completed in 3.55 ms for C:\...\src\Sdk\Sdk.csproj.
1>Restore completed in 2.4 ms for C:\...\src\WpfApp\WpfApp.csproj.
1>Could not find project item with item type 'ProjectReference' and include value '..\Sdk\Sdk.csproj'.
========== Publish: 0 succeeded, 1 failed, 0 skipped ==========
========== Package: 0 succeeded, 1 failed ===========
```
The `Sdk` project is a .NET Core 3.0 class library that contains references to WPF's `FrameworkElement`, which itself requires that I reference the following two packages:
```
<PackageReference Include="Microsoft.NET.Sdk.WindowsDesktop" Version="3.0.0-preview3-27504-2" />
<PackageReference Include="Microsoft.WindowsDesktop.App" Version="3.0.0-preview7-27912-14" />
```
The `WpfApp` project is a .NET Core 3.0 Windows application that has a project reference to the `Sdk` project:
```
<ProjectReference Include="..\Sdk\Sdk.csproj" />
```
When I run `WpfApp` in Visual Studio, the app works perfectly.

I tried changing the two MSBuild output settings to `Diagnostic` but there was no meaningful output in addition to what I pasted above.

I've tried removing projects from the solution and re-adding them and re-adding project references but nothing has helped. Unfortunately, the solution contains proprietary code that I am not allowed to post publicly.

Any idea what's going on?

**Comment by nathan-alden-hp** [comment]
FWIW, I was able to use a combination of `makeappx.exe` and `signtool.exe` to bypass this issue. That makes me think it's a problem with the Visual Studio tooling.

**Comment by carlossanlop** [comment]
@livarcocc @nguerrera can you take a look in case this is an MSBuild problem?

**Comment by nguerrera** [comment]
Don't PackageReference these:
``` xml
<PackageReference Include="Microsoft.NET.Sdk.WindowsDesktop" Version="3.0.0-preview3-27504-2" />
<PackageReference Include="Microsoft.WindowsDesktop.App" Version="3.0.0-preview7-27912-14" />
```

It's a bug that the first one is even on NuGet I think, and the second one should have given you an error. I will follow up on both.

You should have this instead for your WPF class library, as also seen via `dotnet new wpflib` or equivalent File -> New in VS:
```
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <UseWPF>true</UseWPF>
  </PropertyGroup>

</Project>
```

**Comment by nguerrera** [comment]
I don't know that above is related to the issue. I will need to try and repro it.

**Comment by nathan-alden-hp** [comment]
I'm trying this proposal now. Will report back shortly.

**Comment by nathan-alden-hp** [comment]
I remember that I tried this before. It doesn't work. The compiler can't find `FrameworkElement` and I get a confusing warning, as well:
```
Error	CS0246	The type or namespace name 'FrameworkElement' could not be found (are you missing a using directive or an assembly reference?)	Sdk	...\src\Sdk\IUiModule.cs
Error	CS0246	The type or namespace name 'FrameworkElement' could not be found (are you missing a using directive or an assembly reference?)	Sdk	...\src\Sdk\UiModule.cs
Error	CS0246	The type or namespace name 'FrameworkElement' could not be found (are you missing a using directive or an assembly reference?)	Sdk	...\src\Sdk\UiModule.cs
Warning	NETSDK1107	Microsoft.NET.Sdk.WindowsDesktop is required to build Windows desktop applications. 'UseWpf' and 'UseWindowsForms' are not supported by the current SDK.	Sdk	C:\Program Files\dotnet\sdk\3.0.100-preview7-012821\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets	340	
```
My project file:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <UseWpf>true</UseWpf>
    <AssemblyName>Sdk</AssemblyName>
    <RootNamespace>Sdk</RootNamespace>
    <Authors>...</Authors>
    <Product>...</Product>
    <Nullable>enable</Nullable>
    <LangVersion>8.0</LangVersion>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Serilog" Version="2.8.0" />
  </ItemGroup>
</Project>
```

**Comment by nathan-alden-hp** [comment]
I'll also stress that the only way I've been able to get this to work in any fashion (using CLI tools as mentioned earlier) was to include those two `<PackageReference>` elements. Whether it's right or wrong, I can't say.

**Comment by nguerrera** [comment]
You need this additional change to the project file

``` diff
- <Project Sdk="Microsoft.NET.Sdk">
+ <Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
```

That warning NETSDK1107 is trying to alert you to that, but it could be worded better.

**Comment by nathan-alden-hp** [comment]
That appears to have worked! Thanks!

So really, this just comes down to the fact that VS 2019 doesn't yet have the correct project templates for what I'm trying to do.

**Comment by nathan-alden-hp** [comment]
Whoops; closed the issue too quickly.

Unfortunately my warning is still present even after verifying all my projects now use `<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">`:
```
1>Could not find project item with item type 'ProjectReference' and include value '..\Sdk\Sdk.csproj'.
```

**Comment by nguerrera** [comment]
@MSLukeWest can you look at this?

**Comment by sbanni** [comment]
@nguerrera @mslukewest This issue is fixed in 16.3 Preview 2.

**Comment by nathan-alden-hp** [comment]
I just installed .NET Core 3.0 Preview 8 and VS 2019 Preview 16.3.0 Preview 2 and can confirm the issue is resolved. Thank you!
